test/e2e/reconciler/cluster/syncerheartbeat_test.go

```diff
@@ -0,0 +1,128 @@
+)
+
+func TestSyncerHeartbeat(t *testing.T) {
+	t.Parallel()
+
+	source := framework.SharedKcpServer(t)
+
+	t.Log("Creating an organization")
+	orgClusterName := framework.NewOrganizationFixture(t, source)
+
+	t.Log("Creating a workspace")
+	wsClusterName := framework.NewWorkspaceFixture(t, source, orgClusterName, "Universal")
+
+	t.Log("Creating a fake workload server")
+	sink := framework.NewFakeWorkloadServer(t, source, orgClusterName)
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
+	t.Log("Installing test CRDs into source cluster...")
+
+	sourceConfig := source.DefaultConfig(t)
+
+	sourceCrdClusterClient, err := apiextensionsclientset.NewClusterForConfig(sourceConfig)
+	require.NoError(t, err)
+	sourceCrdClient := sourceCrdClusterClient.Cluster(wsClusterName)
+	err = configcrds.Create(ctx, sourceCrdClient.ApiextensionsV1().CustomResourceDefinitions(),
+		metav1.GroupResource{Group: workload.GroupName, Resource: "workloadclusters"},
+		// TODO(marun) Are these necessary for the syncer
+		metav1.GroupResource{Group: apiresource.GroupName, Resource: "apiresourceimports"},
+		metav1.GroupResource{Group: apiresource.GroupName, Resource: "negotiatedapiresources"},
+	)
+	require.NoError(t, err)
+	// TODO(marun) Is this necessary?
+	fixturewildwest.Create(t, sourceCrdClient.ApiextensionsV1().CustomResourceDefinitions(), metav1.GroupResource{Group: wildwest.GroupName, Resource: "cowboys"})
+
+	t.Log("Installing test CRDs into sink cluster...")
+
+	sinkConfig := sink.DefaultConfig(t)
+
+	sinkCrdClient, err := apiextensionsclientset.NewForConfig(sinkConfig)
+	require.NoError(t, err)
+	fixturewildwest.Create(t, sinkCrdClient.ApiextensionsV1().CustomResourceDefinitions(), metav1.GroupResource{Group: wildwest.GroupName, Resource: "cowboys"})
+
+	sourceKcpClusterClient, err := kcpclient.NewClusterForConfig(sourceConfig)
+	require.NoError(t, err)
+	kcpClient := sourceKcpClusterClient.Cluster(wsClusterName)
+
+	t.Log("Creating workload cluster...")
+	workloadCluster, err := framework.CreateCluster(t, source.Artifact, kcpClient, sink)
+	require.NoError(t, err)
+
+	waitForClusterReadyReason(t, ctx, kcpClient.WorkloadV1alpha1().WorkloadClusters(), workloadCluster.Name, workloadv1alpha1.ErrorHeartbeatMissedReason)
+
+	t.Log("Creating workload cluster...")
+	framework.StartWorkspaceSyncer(t, ctx, sets.NewString("cowboys.wildwest.dev"), workloadCluster, source, sink)
+
+	waitForClusterReadyReason(t, ctx, kcpClient.WorkloadV1alpha1().WorkloadClusters(), workloadCluster.Name, "")
+}
+
+func waitForClusterReadyReason(
+	t *testing.T,
+	ctx context.Context,
+	client workloadclient.WorkloadClusterInterface,
+	clusterName string,
+	reason string,
+) {
+	t.Logf("Waiting for cluster %q condition %q to have reason %q", clusterName, workloadv1alpha1.WorkloadClusterReadyCondition, reason)
+	require.Eventually(t, func() bool {
+		cluster, err := client.Get(ctx, clusterName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
```

Comment:
t.Fail

Reply:
I don't think that's kosher i na require loop?